src/architecture/crates.md

```diff
@@ -0,0 +1,23 @@
+# Architecture overview
+
+The servo browser owns a couple of crates that are either mostly independent or based on some upstream project. This gives a list of them.
+We highlight crates with **crate** for crates that are used throughout the rust ecosystem and _crate_ for crates used in Firefox work.
+You should use a bit more care before changing them.
+
+- servo: The main browser
```

Comment:
This PR talks about crates, but this appears to be a list of github repositories. I think this would benefit from the following changes:
* replace all discussion of crates with repositories
* link to each repository
* separate lists for primarily shared with/forked from Firefox (stylo, mozjs, webrender, rust-url, euclid, unicode-bidi), forks of other repositories (rust-content-security-policy), Servo-specific (ci-runners, book, media, servo, malloc_size_of, wpt), and widely-used (html5ever, string-cache)